### PR TITLE
Decouple invite from the creation of a PM room

### PR DESCRIPTION
**Pull request by jaller94**
Issue: https://github.com/matrix-org/matrix-appservice-irc/issues/1254#issuecomment-788156440

Split of from: https://github.com/matrix-org/matrix-appservice-irc/pull/1282#discussion_r609654092

### CODE REVIEW

File: src/bridge/IrcHandler.ts [hunk 1]
```diff
@@ -167,7 +169,6 @@ export class IrcHandler {
                         visibility: "private",
                         // We deliberately set our own power levels below.
                         // preset: "trusted_private_chat",
-                        invite: [toUserId],
```
Comment (Half-Shot) [inline src/bridge/IrcHandler.ts:170]: Hold on a sec, are we actually inviting the user at all now?

**Review by Half-Shot — COMMENTED** [review]
I think this is okay, but are we inviting the user?

File: src/bridge/IrcHandler.ts (hunk 1 above)
Comment (jaller94) [reply src/bridge/IrcHandler.ts:170]: `ensureMatrixUserJoined` should join the user right after the creation.

https://github.com/matrix-org/matrix-appservice-irc/blob/7dc74b92cd6906619e53c262ea14682fd901fb8f/src/bridge/IrcHandler.ts#L308-L311

Comment (jaller94) [reply src/bridge/IrcHandler.ts:170]: Works for me.
## Test plan
1. Rebuild & Restart bridge with this branch
2. As an IRC user, message a Matrix user.
3. Use `/nick new-name` to a new username.
4. Message the Matrix user again.

https://user-images.githubusercontent.com/10872136/116164333-e260e300-a6f9-11eb-968f-4f6beea07970.mp4

**Review by Half-Shot — APPROVED** [review]
I think this looks great!

**Pull request merged**